### DISCUSSION TITLE
Failing to install because of byte compile error: `void-function static-if`

### DISCUSSION THREAD

**Discussion by nickjmeyer** [magit/magit#5278] [Support] [answered]
I'm running into issues with installing `magit`.  To reproduce, I'm running with `emacs -nw -Q` to skip all my initialization.  When I run `M-x package-install` `magit`, I get the following errors:
```
Compiling file /home/nickjmeyer/.emacs.d/elpa/transient-0.8.1/transient.el at Thu Dec 26 19:41:36 2024
Entering directory ‘/home/nickjmeyer/.emacs.d/elpa/transient-0.8.1/’
transient.el:1229:8: Error: Symbol’s function definition is void: static-if
```

The above error is in `transient`.  But there are a bunch of other failures for the same error in different files within `magit` itself.

From some research, it seems `static-if` is provided by `compat`.  I tried to explicitly install `compat` first, but I got the same errors.

Here's my emacs version info: 
```
GNU Emacs 30.0.50 (build 1, x86_64-pc-linux-gnu, GTK+ Version 3.24.24, cairo version 1.16.0) of 2023-02-15, unofficial emacs-snapshot build: http://emacs.ganneff.de/, git commit 6a677fd3b6677248a8bc1787b105879684289fa2
```
Unfortunately `magit-version` won't work because I can't get it installed :(

Anyone have any ideas?

**Comment by kyleam** [comment, marked as answer]
* Your version info indicates you're using an Emacs built from 6a677fd3b66 (2023-02-15).

 * `static-if` was added in 97652d0e7a5 (New conditional compilation macro static-if, 2023-09-03).

 * IIUC [compat doesn't load its compatibility code][1], including `static-if`, because your Emacs version is at least 30.0.50.

To fix the issue, I'd suggest 1) updating to a recent developmental build or 2) downgrading to the latest release (Emacs 29.4).

[1]: https://github.com/emacs-compat/compat/blob/a415c5f47a14355325d72a3bab8b62ddaebe0f10/compat.el#L51-L56

**Reply by nickjmeyer** [reply]
Well sure enough that was it.  Thanks!
